cucumber/api/features/login.feature

```diff
@@ -5,12 +5,93 @@ Feature: Exchange a role's password for its API key
 
   Background:
     Given I create a new user "alice"
+    And I have host "app"
 
   Scenario: Password can be used to obtain API key
     Given I set the password for "alice" to "My-Password1"
     When I can GET "/authn/cucumber/login" with username "alice" and password "My-Password1"
     Then the HTTP response content type is "text/plain"
     And the result is the API key for user "alice"
+    And there is an audit record matching:
+    """
+      <84>1 * * conjur * authn
+      [auth@43868 authenticator="authn" service="default:webservice:conjur/authn user="cucumber:user:alice"]
+      [subject@43868 role="cucumber:user:alice"]
+      [action@43868 operation="login" result="success"]
+      cucumber:user:alice successfully logged in with authenticator authn service default:webservice:conjur/authn
+    """
+
+
+  Scenario: Password can be used by host to obtain API key
+    Given I set the password for "host/app" to "My-Password1"
+    When I GET "/authn/cucumber/login" with username "host/app" and password "My-Password1"
+    Then the HTTP response content type is "text/plain"
+    And the result is the API key for host "app"
+    And there is an audit record matching:
+    """
+      <84>1 * * conjur * authn
+      [auth@43868 authenticator="authn" service="default:webservice:conjur/authn user="cucumber:host:app"]
+      [subject@43868 role="cucumber:host:app"]
+      [action@43868 operation="login" result="success"]
+      cucumber:host:app successfully logged in with authenticator authn service default:webservice:conjur/authn
+    """
+
+  Scenario: Wrong password cannot be used to obtain API key
+    When I GET "/authn/cucumber/login" with username "alice" and password "Wrong-Password"
+    Then the HTTP response status code is 401
+    And there is an audit record matching:
+    """
+      <84>1 * * conjur * authn
+      [subject@43868 role="cucumber:user:alice"]
+      [auth@43868 authenticator="authn" service="cucumber:webservice:conjur/authn" user="not-found"]
```

Comment:
i may be missing something but don't we know that the user is `cucumber:user:alice` in this case? why do we need the correct password to write an audit log with the user that tried to login?

Reply:
@netacoral is there a product reason for this? why do we write the user id to the audit log only if the request succeeded?

Reply:
@orenbm, speaking with @netacoral, the masking of the user is going to be only when user doesn't exist, not for security purposes but rather to avoid trashing the audit with many (possibly different) non-existing usernames. 
Thus, my latest commit adds a fix that only masks the username in case it doesn't exist